pkg/apis/postgres-operator.crunchydata.com/v1beta1/postgrescluster_types.go

```diff
@@ -478,6 +478,23 @@ type PostgresInstanceSetSpec struct {
 	// More info: https://www.postgresql.org/docs/current/wal.html
 	// +optional
 	WALVolumeClaimSpec *corev1.PersistentVolumeClaimSpec `json:"walVolumeClaimSpec,omitempty"`
+
+	// The list of tablespaces volumes to mount for this postgrescluster
+	// This function is currently in alpha as of PGO v5.4
+	// +optional
+	TablespaceVolumes []TablespaceVolume `json:"tablespaceVolumes,omitempty"`
+}
+
+type TablespaceVolume struct {
+	// The name for the tablespace, used as the path name for the volume.
+	// Must be unique in the instance set since they become the directory names.
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
```

Comment:
🤔 This value goes multiple places and should probably have validation to ensure those succeed.

- This value goes into a PVC name. I recall [PostgresUserSpec](https://github.com/CrunchyData/postgres-operator/blob/8b14582aaa9df139b8eb38fe48a47f546db61f02/pkg/apis/postgres-operator.crunchydata.com/v1beta1/postgres_types.go#L41) did similar considering Secret names.

- This value goes into a Volume name. [Docs for that](https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/volume/#Volume) say "DNS label," but K8s defines more than one of those. If you track down the exact validation, I would be grateful to see it explained in `internal/naming/limitations.md`. 🙇🏻 

- This value goes into a directory name, so no slashes? I don't know what else to consider here.
  🤔 `fsck` `lost+found`...?

(Other places?)

Reply:
Oof, good thinking. How do you feel about `^[a-zA-Z0-9]+$`? Like, what's the upshot to being permissive?

Reply:
> How do you feel about `^[a-zA-Z0-9]+$`?

["DNS labels"](https://github.com/CrunchyData/postgres-operator/blob/master/internal/naming/limitations.md#dns-label) need to be lowercase, and some need to start with a letter. Could `^[a-z][a-z0-9]*$` to cover both?

I also had yet to consider empty string. Good catch. 🤔 I don't know if "min length" validation has a better message or is overkill or what.

> Like, what's the upshot to being permissive?

Dunno. I do know we can always go from less-permissive now to more-permissive later. (We should not do the reverse.) That is, objects stored in the API (that are/were valid) should/must continue to be valid in the future.

Reply:
OK, updated: alphanumeric, lowercase only, alphabet to start. And added minLength for that clearer error message.